src/transformStateWithClones.js

```diff
@@ -7,7 +7,34 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const resultArr = [];
+
+  const obj = { ...state };
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties':
+        Object.assign(obj, action.extraData);
+        resultArr.push({ ...obj });
```

Comment:
No need to `push` in each case. Write it once in the end of iteration(after `switch case`).